src/CoreArbiterServer.cc

```diff
@@ -1067,8 +1067,12 @@ CoreArbiterServer::distributeCores()
                     if (!sendData(thread->socket, &core->id, sizeof(int),
                                   "Error sending core ID to thread " +
                                         std::to_string(thread->id))) {
-                        exit(-1);
-                        return;
+                        // TODO(Qian): is it a good solution to ignore error?
```

Comment:
After thinking about this some more, I'd still be much more comfortable if we can do the cleanup here; otherwise there will be a window of vulnerability where the Core Arbiter Server incorrectly believes that a thread is assigned to a core (thus preventing that core from being used by any other thread) when the reality is that the thread is already dead.